src/OSPSuite.Presentation/Presenters/Charts/DataBrowserPresenter.cs

```diff
@@ -155,6 +168,20 @@ public void UsedChangedFor(DataColumnDTO dataColumnDTO, bool used)
       {
          raiseUsedChanged(dataColumnDTO.DataColumn, used);
          updateDataSelection(_view.SelectedColumns);
+         updateLinkedObservedData(dataColumnDTO.DataColumn, used);
+      }
+
+      private void updateLinkedObservedData(DataColumn dataColumn, bool used)
+      {
+         if (!_linkMappedOutputs) return;
+
+         var updatedDataColumnName = dataColumn.PathAsString;
+         if (!OutputMappingNames.Contains(updatedDataColumnName)) return;
+
+         var linkedObservedDataNames = OutputMappingNames[updatedDataColumnName];
+         var linkedObservedData = _dataColumnDTOCache.KeyValues.Where(x => linkedObservedDataNames.Contains(x.Key.Repository.Name)).Select(x => x.Value).ToList();
```

Comment:
This is what I mean here. Create a method for this. getObservedDataLinkedToOuptut or something like this.